packages/flutter_tools/lib/src/windows/windows_version_validator.dart

```diff
@@ -86,7 +89,17 @@ class WindowsVersionValidator extends DoctorValidator {
     if (matches.length == 1 &&
         !kUnsupportedVersions.contains(matches.elementAt(0).group(1))) {
       windowsVersionStatus = ValidationType.success;
-      statusInfo = 'Installed version of Windows is version 10 or higher';
+      final WindowsVersionExtractionResult details = await _versionExtractor.getDetails();
+      String? caption = details.caption;
+      if (caption == null || caption.isEmpty) {
+        final bool isWindows11 = int.parse(matches.elementAt(0).group(3)!) > 20000;
+        if (isWindows11) {
+          caption = 'Windows 11 or higher';
+        } else {
+          caption = 'Windows 10';
+        }
+      }
+      statusInfo = '$caption, ${details.displayVersion}, ${details.releaseId}';
```

Comment:
If the `reg` call fails, this will print something like `Windows 11 or higher, null, null`. I added a failing test to cover this behavior. Perhaps we could do `<String>[caption, details.displayVersion, details.releaseId].join(', ')` instead? With this, we should just see `Windows 11 or higher`.

Reply:
That's a good idea. I was not aware that the join method excludes `null`s

Reply:
Actually it does not work. when I run `print(<String?>['hallo', null, 'world'].join(' '));` I get the output "hello null world", so those need to filtered out first

Reply:
> Actually it does not work. when I run `print(<String?>['hallo', null, 'world'].join(' '));` I get the output "hello null world", so those need to filtered out first

Oh yeah, I forgot that part. Sorry 😓